### PR TITLE
[FIX] tests: properly log errors

### PR DESCRIPTION
**Pull request by mart-e**
If a test is failing with `assertRecordValues`, the comparison and the error
logging were not computed the same way.
The `_compare_candidate` was accessing the record via
```python
record_value = record[field_name]
```
which uses the cache

While the `_format_message` was doing
```python
all_records_values = records.read(list(expected_values[0].keys()), load=False)
```
which queries directly the database.

If the test was failing (correctly) because of a wrong cache invalidation
(e.g. update the value directly in SQL without invalidating the cache),
the error message would look like:
```
AssertionError: Record doesn't match expected values at index 0.
Current values:
{'id': 42, 'name': 'Foo'}
Expected values:
{'name': 'Foo'}
```

Example of failing test:
```python
    def test_failure(self):
        categroise = self.env['res.partner.category'].create({'name': 'A Category'})
        self.assertRecordValues(categroise, [{'name': 'A Category'}])
        self.env.cr.execute("UPDATE res_partner_category SET name=%s WHERE id=%s", ('Other Category', categroise.id))
        self.assertRecordValues(categroise, [{'name': 'Other Category'}])
```

### CODE REVIEW

File: odoo/tests/common.py [hunk 1]
```diff
@@ -334,9 +334,23 @@ def _compare_candidate(record, candidate):
                     return False
             return True
 
+        def _repr_field_value(record, field_name):
+            record_value = record[field_name]
+            field_type = record._fields[field_name].type
+            if field_type == 'monetary':
+                currency_field_name = record._fields[field_name].currency_field
+                record_currency = record[currency_field_name]
+                return record_currency and record_currency.amount_to_text(record_value) or record_value
```
Comment (rco-odoo) [inline odoo/tests/common.py]: `amount_to_text` ???

Comment (mart-e) [reply odoo/tests/common.py]: Right, `round` is probably closer to what we are looking for (replicate `record_currency.compare_amounts(candidate_value, record_value)`)

**Comment by mart-e** [comment]
robodoo r+

**Comment by mart-e** [comment]
@robodoo retry please

**Pull request closed without merging**